src/platform/activation_abi.cpp

```diff
@@ -0,0 +1,69 @@
+#include "pal_internal.h"
+#include "opaque_string_wrapper.h"
+#include "platform_activation.h"
+
+namespace xlang::impl
+{
+    template <typename char_type>
+    void* get_activation_factory(
+        xlang_string class_name,
+        xlang_guid const& iid)
+    {
+        for (auto current_namespace = enclosing_namespace(to_string_view<char_type>(class_name));
```

Comment:
Apparently, here you try to get the enclosing namespace, by cutting the class name from `class_name` right away.

But [XDN03 - Platform Support for Activation Factories](/Microsoft/xlang/blob/master/design_notes/XDN03%20-%20Platform%20Support%20for%20Activation%20Factories.md) design note has the following example:

> ...imagine an app requests a factory for the class *MyComponent.Feature.Widget*...
> * The PAL searches for the library *MyComponent.Feature.Widget.(dll/so)* (on Windows/Linux, respectively).
> * This library is not found in the application's directory, so it moves up one namespace level ...

Is it the design note that should be updated, or should you initialize `current_namespace` with just `to_string_view<char_type>(class_name)`?

Reply:
Good catch @Alovchin91. @DefaultRyan is out of the office currently so I'm finishing off this PR. I'm pretty sure XDN03 is wrong here, but I opened #55 for @DefaultRyan to look at when he's back in the office.